### PR TITLE
Update reverse proxy docs with what we've learned from #17986

### PR DESCRIPTION
**Pull request by MadLittleMods**
Update reverse proxy docs with what we've learned from https://github.com/element-hq/synapse/pull/17986

Also vice versa and update our nginx config with what I learned from the reverse proxy docs.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] [Code style](https://element-hq.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://element-hq.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

File: docker/conf-workers/nginx.conf.j2 [hunk 1]
```diff
@@ -38,6 +38,9 @@ server {
 {% if using_unix_sockets %}
         proxy_pass http://unix:/run/main_public.sock;
 {% else %}
+        # note: do not add a path (even a single /) after the port in `proxy_pass`,
+        # otherwise nginx will canonicalise the URI and cause signature verification
+        # errors.
```
Comment (MadLittleMods) [inline docker/conf-workers/nginx.conf.j2:43]: This comment comes from our [`reverse_proxy.md` docs](https://github.com/element-hq/synapse/blob/a00d0b3d0e72cd56733c30b1b52b5402c92f81cc/docs/reverse_proxy.md#L71-L73).

What I think it actually means -> https://stackoverflow.com/questions/16157893/nginx-proxy-pass-404-error-dont-understand-why/62753212#62753212

File: docs/reverse_proxy.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ server {
         proxy_pass http://localhost:8008;
         proxy_set_header X-Forwarded-For $remote_addr;
         proxy_set_header X-Forwarded-Proto $scheme;
-        proxy_set_header Host $host;
+        proxy_set_header Host $host:$server_port;
```
Comment (MadLittleMods) [inline docs/reverse_proxy.md:77]: Updating our docs to match what we fixed in https://github.com/element-hq/synapse/pull/17986

**Review by anoadragon453 — APPROVED** [review]
Hopefully this will save folks quite a lot of time in the future. Thanks for carrying this through to the end!

**Comment by MadLittleMods** [comment]
Thanks for the review and merge @anoadragon453 🐍

**Pull request merged**